File: solutions/ruby/matrix/2/matrix.rb
class Matrix

  def rows
    @matrix.split(/\n/).map { |row| row.split.map(&:to_i) }
  end

  def columns
    columns = []
    
    rows.map(&:size).max.times do |i|
      columns << rows.map { |row| row[i] }
    end

    columns
  end

  private

  def initialize(matrix)
    @matrix = matrix
  end
  
end

File: solutions/python/matrix/2/matrix.py
class Matrix:
    def __init__(self, text):
        self.rows = [[int(c) for c in row.split()] for row in text.split('\n')]

    def row(self, index):
        return self.rows[index-1]

    def column(self, index):
        return [cell[index-1] for cell in self.rows]
